### PR TITLE
[Port]Recover the correct axis in the parent layout for reduce ops 

### PR DESCRIPTION
**Pull request by Stonepia**
https://github.com/openai/triton/pull/1885

### CODE REVIEW

**Comment by EikanWang** [comment]
Pls. suspense this PR merge before the e2e ci is ready.

File: lib/Conversion/TritonGPUToSPIRV/ReduceOpToSPIRV.cpp [hunk 1]
```diff
@@ -87,13 +86,17 @@ struct ReduceOpSPIRVConversion
...
                           SmallVector<Value> &writeIdx,
-                          std::map<int, Value> &ints, unsigned axis) const {
+                          std::map<int, Value> &ints, unsigned originalAxis,
+                          unsigned axis) const {
     if (auto sliceLayout = layout.dyn_cast<SliceEncodingAttr>()) {
-      auto dim = sliceLayout.getDim();
-      assert(dim != axis && "Reduction axis cannot be sliced");
+      // auto dim = sliceLayout.getDim();
+      // Recover the axis in the parent layout
+      // std::cout << " dim " << dim << " axis " << axis << "\n";
```
Comment (chengjunlu) [inline lib/Conversion/TritonGPUToSPIRV/ReduceOpToSPIRV.cpp:94]: Remove debug code.

Comment (Stonepia) [reply lib/Conversion/TritonGPUToSPIRV/ReduceOpToSPIRV.cpp]: Thanks! Resolved in https://github.com/intel/intel-xpu-backend-for-triton/pull/53/commits/54e7731d300f3a32a9f155655cef561c79fafeb8

**Comment by Stonepia** [comment]
Hello @EikanWang, This PR needs to be first merged before CI is set up. It is because the test of `test_chain_reduce` will trigger a segmentation fault and block the PR process.

**Review by chengjunlu — APPROVED** [review]

**Review by EikanWang — APPROVED** [review]

**Pull request merged**